### PR TITLE
DOC: Adding Cylon under ecosystem/out of core

### PR DESCRIPTION
**Pull request by chathurawidanage**
- [x]  Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

This PR adds a link and a description about Cylon to the ecosystem page.  

Github: [https://github.com/cylondata/cylon](https://github.com/cylondata/cylon)
Web: [https://cylondata.org/](https://cylondata.org/)

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/ecosystem.rst [hunk 1]
```diff
@@ -405,6 +405,11 @@ Blaze provides a standard API for doing computations with various
 in-memory and on-disk backends: NumPy, pandas, SQLAlchemy, MongoDB, PyTables,
 PySpark.
 
+`Cylon <https://cylondata.org/>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Cylon is a fast, scalable, distributed memory parallel runtime with a Pandas like Python DataFrame API. ”Core Cylon” is implemented with C++ using Apache Arrow format to represent the data in-memory. Cylon DataFrame API implements most of the core operators of Pandas such as merge, filter, join, concat, group-by, drop_duplicates, etc. These operators are designed to work across thousands of cores to scale applications. It can interoperate with Pandas DataFrame by reading data from Pandas or converting data to Pandas so users can selectively scale parts of their Pandas DataFrame applications.
```
[the remark this answers is not in the document]
Comment (chathurawidanage) [reply doc/source/ecosystem.rst]: Added an example.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks for the update @chathurawidanage

File: doc/source/ecosystem.rst [hunk 2]
```diff
@@ -405,6 +405,35 @@ Blaze provides a standard API for doing computations with various
...
+    # Initialize Cylon distributed environment 
+    config: MPIConfig = MPIConfig()
+    env: CylonEnv = CylonEnv(config=config, distributed=True)
+
+    df1: DataFrame = read_csv('/tmp/csv1.csv')
+    df2: DataFrame = read_csv('/tmp/csv2.csv')
+    
+    # Using 1000s of cores across the cluster to compute the join
+    df3: Table = df1.join(other=df2, on=[0], algorithm="hash", env=env)
+
```
Comment (datapythonista) [inline doc/source/ecosystem.rst:434]: I think you've got spaces in this blank line, you'll need to remove them to get the CI green.

**Comment by datapythonista** [comment]
CI still red for trailing spaces, `git diff upstream/master` should highlight them if your editor doesn't.

**Comment by chathurawidanage** [comment]
Thanks, @datapythonista that did the trick!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @chathurawidanage

**Pull request merged**